packages/core/utils/src/traverse/query-populate.ts

```diff
@@ -24,53 +24,63 @@ const isKeyword = (keyword: string) => {
   };
 };
 
+const isWildcard = (value: unknown): value is '*' => value === '*';
+
+const isPopulateString = (value: unknown): value is string => {
+  return isString(value) && !isWildcard(value);
+};
+
 const isStringArray = (value: unknown): value is string[] =>
   isArray(value) && value.every(isString);
 
 const isObj = (value: unknown): value is Record<string, unknown> => isObject(value);
 
 const populate = traverseFactory()
-  // Array of strings ['foo', 'foo.bar'] => map(recurse), then filter out empty items
+  .intercept(isPopulateString, async (visitor, options, populate, { recurse }) => {
+    const populateObject = pathsToObjectPopulate([populate]);
+    const traversedPopulate = (await recurse(visitor, options, populateObject)) as PopulateObject;
+    const [result] = objectPopulateToPaths(traversedPopulate);
```

Comment:
Could we add some context to this bit to make sure it's not altered by future contributors? Basically explaining why we have to transform in & out the strings